plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/utils/CodeTransformApiUtils.kt

```diff
@@ -144,6 +172,119 @@ suspend fun JobId.pollTransformationStatusAndPlan(
+    getLogger<CodeModernizerManager>().info { "About to copy the original project ZIP to: $copyOfProjectSources" }
+    val originalProjectZip = CodeModernizerManager.getInstance(project).codeTransformationSession?.sessionContext?.originalUploadZipPath
+    originalProjectZip?.let { unzipFile(it, copyOfProjectSources, isSqlMetadata = false, extractOnlySources = true) }
+    copyOfProjectSources = copyOfProjectSources.resolve("sources") // where the user's source code is within the upload ZIP
+    getLogger<CodeModernizerManager>().info { "Copied and unzipped original project sources to: $copyOfProjectSources" }
+
+    val targetDir = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(copyOfProjectSources.toFile())
+        ?: throw RuntimeException("Cannot find copy of project sources directory")
+
+    withContext(EDT) {
+        runWriteAction {
+            // create temp module with project copy so that we can apply diff.patch
+            val modifiableModel = ModuleManager.getInstance(project).getModifiableModel()
+            val tempModule = modifiableModel.newModule(
+                Paths.get(targetDir.path).resolve("temp.iml").toString(),
+                JavaModuleType.getModuleType().id
+            )
+
+            try {
+                val moduleModel = ModuleRootManager.getInstance(tempModule).modifiableModel
+                moduleModel.addContentEntry(targetDir.url)
+                moduleModel.commit()
+                modifiableModel.commit()
+
+                // apply diff.patch
+                val patchReader = PatchReader(clientInstructionsPath)
+                patchReader.parseAllPatches()
+                PatchApplier(
+                    project,
+                    targetDir,
+                    patchReader.allPatches,
+                    null,
+                    null
+                ).execute()
+                getLogger<CodeModernizerManager>().info { "Successfully applied patch file at $clientInstructionsPath" }
+
+                val virtualFile = LocalFileSystem.getInstance().findFileByIoFile(clientInstructionsPath.toFile())
+                    ?: throw RuntimeException("Cannot find patch file at $clientInstructionsPath")
+                FileEditorManager.getInstance(project).openFile(virtualFile, true)
+            } catch (e: Exception) {
+                getLogger<CodeModernizerManager>().error {
+                    "Error applying intermediate diff.patch for job ${jobId.id} and artifact $artifactId located at " +
+                        "$clientInstructionsPath: $e"
+                }
+            } finally {
+                runWriteAction {
+                    ModuleManager.getInstance(project).disposeModule(tempModule)
+                }
+            }
+        }
+    }
+
+    val (exitCode, buildOutput) = runClientSideBuild(targetDir, CodeModernizerManager.LOG, project)
+    getLogger<CodeModernizerManager>().info { "Ran client-side build with an exit code of $exitCode" }
```

Comment:
can you extract this a private val
```suggestion
    getLogger<CodeModernizerManager>().info { "Ran client-side build with an exit code of $exitCode" }
```